crates/red_knot/docs/tracing.md

```diff
@@ -0,0 +1,103 @@
+# Tracing
```

Comment:
I assume that the `crates/red_knot/docs/...` directory is for developer-facing docs, not user-facing docs?

Reply:
For now, that's certainly true I don't know about long-term. We could have `docs/users` and `docs/dev` if we end up with needing both. But most user-facing documentation should probably live on the website.